test/DefaultCluster.Tests/GrainReferenceTest.cs

```diff
@@ -111,22 +111,22 @@ public void GrainReference_Json_Serialization()
             await grain.SetA(56820);
             var input = new GenericGrainReferenceHolder
             {
-                Reference = grain
+                Reference = grain as GrainReference
             };
 
             var json = JsonConvert.SerializeObject(input, settings);
             var output = JsonConvert.DeserializeObject<GenericGrainReferenceHolder>(json, settings);
 
             Assert.Equal(input.Reference, output.Reference);
-            var reference = output.Reference.Cast<ISimpleGrain>();
-            Assert.Equal(56820, await reference.GetA());
+            var reference = output.Reference;
+            Assert.Equal(56820, await ((ISimpleGrain)reference).GetA());
         }
 
         [Serializable]
         public class GenericGrainReferenceHolder
         {
             [JsonProperty]
-            public ISimpleGrain Reference { get; set; }
+            public GrainReference Reference { get; set; }
```

Comment:
yeah, I think it makes sense to have both, but it's up to you. There might already be coverage for the other case